app/controllers/api/events_controller.rb

```diff
@@ -1,4 +1,15 @@
 class API::EventsController < API::RestfulController
+  include DashboardHelper
+
+  def dashboard_by_date
+    @events = page_collection dashboard_events
+    respond_with_collection
+  end
+
+  def dashboard_by_group
+    @events = grouped dashboard_events.group_by(&:organization_id)
```

Comment:
Grouping on the server side is the only way to guarantee we return the
right amount of threads for each group.

If I have threads A - Z, sorted desc by date, and groups 1-3, and threads
XYZ are all in group 3, I'm not going to return then with the API call
unless I do a grouping server side.

On Wed, Apr 1, 2015 at 5:52 AM, Robert Guthrie <notifications@github.com>
wrote:

> In app/controllers/api/events_controller.rb
> <https://github.com/loomio/loomio/pull/2074#discussion_r27527104>:
>
> > @@ -1,4 +1,15 @@
> >  class API::EventsController < API::RestfulController
> > +  include DashboardHelper
> > +
> > +  def dashboard_by_date
> > +    @events = page_collection dashboard_events
> > +    respond_with_collection
> > +  end
> > +
> > +  def dashboard_by_group
> > +    @events = grouped dashboard_events.group_by(&:organization_id)
>
> I'm starting to think that these should be the same API call.. I don't
> understand why you group on the server side.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2074/files#r27527104>.
>